app/views/linkers/index.html.erb

```diff
@@ -8,71 +8,61 @@
-  var th2 = document.createElement('th'); th2.appendChild(document.createTextNode('HTTP Code'));
-  var th3 = document.createElement('th'); th3.appendChild(document.createTextNode('HTTP Status'));
-  var th4 = document.createElement('th'); th4.appendChild(document.createTextNode('Redirect'));
-  var th5 = document.createElement('th'); th5.appendChild(document.createTextNode('Search Term Exists?'));
-
-  thdr.appendChild(th1);
-  thdr.appendChild(th2);
-  thdr.appendChild(th3);
-  thdr.appendChild(th4);
-  thdr.appendChild(th5);
-
-  thd.appendChild(thdr);
-  tbl.appendChild(thd);
-  tbl.appendChild(tbdy);
-
-  document.getElementById('testdiv').appendChild(tbl);
-
-  var csv_list = document.getElementById("CSVfile").files[0];
-  var reader = new FileReader();
-
-  reader.onload = function(event) {
-
-    var lines = reader.result.split('\n');
-
-    for(var i = 0; i < lines.length; i+=1) {
-
-      var new_row = document.createElement('tr');
-      var new_cell = document.createElement('td'); new_cell.appendChild(document.createTextNode(lines[i]));
-      new_row.appendChild(new_cell);
-
-      var xhttp = new XMLHttpRequest();
-      xhttp.open("GET", "http://linkfixer.herokuapp.com/ping_url?url_string=" + encodeURIComponent(lines[i]) + "&search_term=" + encodeURIComponent(document.getElementById('search_term').value), false);
-      xhttp.send();
-      if (xhttp.readyState == 4 && xhttp.status == 200) {
-        console.log(xhttp.responseText);
-        new_cell = document.createElement('td'); new_cell.appendChild(document.createTextNode(JSON.parse(xhttp.responseText)[0])); new_row.appendChild(new_cell);
-        new_cell = document.createElement('td'); new_cell.appendChild(document.createTextNode(JSON.parse(xhttp.responseText)[1])); new_row.appendChild(new_cell);
-        new_cell = document.createElement('td'); new_cell.appendChild(document.createTextNode(JSON.parse(xhttp.responseText)[2])); new_row.appendChild(new_cell);
-        new_cell = document.createElement('td'); new_cell.appendChild(document.createTextNode(JSON.parse(xhttp.responseText)[3])); new_row.appendChild(new_cell);
-        tbdy.appendChild(new_row);
-      }
-
-    }
-
-  }
-
-  reader.readAsText(csv_list);
+$(function() {
+  $('#test-button').click(function(e) {
+    e.preventDefault();
+    $('#test-results').show();
+    readCSVFile();
+  });
+
+  var readCSVFile = function() {
+    var csvContents = document.getElementById("CSVfile").files[0],
+        reader = new FileReader(),
+        tbdy = $('#test-results tbody')[0];
+
+    reader.onload = function(event) {
+      var lines = reader.result.split('\n');
+
+      lines.forEach(function(line) {
+        var $tr = $('<tr>'),
+            requestParams = { url_string: line, search_term: $('#search-term').val() };
+
+        $tr.append($('<td>' + line + '</td>'));
+        $.getJSON('<%= ping_url_path %>', requestParams).done(function(result) {
```

Comment:
One benefit of `$.getJSON()` is that you can pass it a params object like I did here, and it will take care of all the URL encoding/etc. for you for free.